src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -2519,3 +2521,69 @@ void ngraph::runtime::cpu::pass::CPUQuantFusion::construct_qconvb_add()
     auto m = std::make_shared<pattern::Matcher>(prelu, "CPUQuantFusion.QConvBiasSignedAdd");
     this->add_matcher(m, callback);
 }
+
+// Convert a QuantizedDot which takes [m,n]*[n,k] to
+// QuantizedMatmul which reorders input1 and does [m,n]*[k,n]
+// which is what mkldnn wants
+void ngraph::runtime::cpu::pass::CPUQuantFusion::construct_quantized_matmul(bool requantize,
+                                                                            bool with_relu)
+{
+    Shape shape_input0{2, 3};
+    Shape shape_input1{3, 4};
+    auto input0 = std::make_shared<pattern::op::Label>(element::u8, shape_input0);
+    auto input1 = std::make_shared<pattern::op::Label>(element::i8, shape_input1);
+    auto scale = std::make_shared<pattern::op::Label>(element::f32, Shape{});
+
+    std::shared_ptr<ngraph::op::Op> q_dot;
+    if (!requantize && !with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, false, false);
+    }
+    else if (!requantize && with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, false, true);
+    }
+    else if (requantize && !with_relu)
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, true, false);
+    }
+    else
+    {
+        q_dot = std::make_shared<ngraph::op::QuantizedDot>(input0, input1, scale, true, true);
+    }
+    auto callback = [input0, input1, scale](pattern::Matcher& m) {
+        NGRAPH_DEBUG << "In callback for Qdot against node = " << m.get_match_root()->get_name();
+        auto pattern_map = m.get_pattern_map();
+
+        auto qdot = std::dynamic_pointer_cast<ngraph::op::QuantizedDot>(m.get_match_root());
+        auto input_0 = pattern_map[input0];
+        auto input_1 = pattern_map[input1];
+        auto scale_new = pattern_map[scale];
+
+        if (input_0->get_element_type() == element::u8 &&
+            input_1->get_element_type() == element::u8)
+        {
+            return false;
+        }
+
+        std::shared_ptr<ngraph::op::Op> qmatmul;
+        if (input_1->get_shape()[0] != input_1->get_shape()[1])
```

Comment:
you will need to always reshape/transpose input_1 irrespective of its shape since QuantizedMatMul expects the reduction axis to be the first axis in input_1 while QuantizedDot has the reduction axis as the second axis.